app/models/concerns/github_repoable.rb

```diff
@@ -8,14 +8,18 @@ def add_team_to_github_repository
     github_repository = GitHubRepository.new(organization.github_client, github_repo_id)
     github_team       = GitHubTeam.new(organization.github_client, github_team_id)
 
-    github_team.add_team_repository(github_repository.full_name)
+    options = {}
+    options[:permission] = 'admin' if give_admin_permission?
+    github_team.add_team_repository(github_repository.full_name, options)
   end
 
   def add_user_as_collaborator
     github_user = GitHubUser.new(user.github_client, user.uid)
     repository  = GitHubRepository.new(organization.github_client, github_repo_id)
 
-    delete_github_repository_on_failure { repository.add_collaborator(github_user.login) }
+    options = {}
+    options[:permission] = 'admin' if give_admin_permission?
+    delete_github_repository_on_failure { repository.add_collaborator(github_user.login, options) }
   end
 
   # Public
```

Comment:
I would rather `give_admin_permission?` be in this module, and in one method.

Maybe something similar to this:

```ruby
def give_admin_permission?
  student_assignment = self.has_attribute?(:assignment) ? assignment : group_assignment
  student_assignment.students_are_repo_admins?
end
```